src/status_im/ui/screens/discover/views.cljs

```diff
@@ -1,95 +1,114 @@
-   [status-im.utils.platform :refer [platform-specific]]
-   [status-im.i18n :refer [label]]
-   [status-im.ui.screens.discover.styles :as st]
-   [status-im.ui.screens.contacts.styles :as contacts-st]))
+    [re-frame.core :refer [dispatch subscribe]]
+    [clojure.string :as str]
+    [status-im.components.react :refer [view
+                                        scroll-view
+                                        text
+                                        text-input
+                                        touchable-highlight]]
+    [status-im.components.icons.vector-icons :as vi]
+    [status-im.components.toolbar-new.view :as toolbar]
+    [status-im.components.toolbar-new.actions :as act]
+    [status-im.components.drawer.view :as drawer]
+    [status-im.components.carousel.carousel :as carousel]
+    [status-im.ui.screens.discover.components.views :as components]
+    [status-im.utils.platform :as platform]
+    [status-im.i18n :as i18n]
+    [status-im.ui.screens.discover.styles :as styles]
+    [status-im.ui.screens.contacts.styles :as contacts-st]))
 
 (defn get-hashtags [status]
   (let [hashtags (map #(str/lower-case (str/replace % #"#" "")) (re-seq #"[^ !?,;:.]+" status))]
     (or hashtags [])))
 
 (defn toolbar-view [show-search? search-text]
-  [toolbar-with-search
+  [toolbar/toolbar-with-search
    {:show-search?       show-search?
     :search-text        search-text
     :search-key         :discover
-    :title              (label :t/discover)
-    :search-placeholder (label :t/search-tags)
+    :title              (i18n/label :t/discover)
+    :search-placeholder (i18n/label :t/search-tags)
     :nav-action         (act/hamburger drawer/open-drawer!)
     :on-search-submit   (fn [text]
                           (when-not (str/blank? text)
                             (let [hashtags (get-hashtags text)]
                               (dispatch [:set :discover-search-tags hashtags])
                               (dispatch [:navigate-to :discover-search-results]))))}])
 
-(defn title [label-kw spacing?]
-  [view st/section-spacing
-   [text {:style      (merge (get-in platform-specific [:component-styles :discover :subtitle])
-                             (when spacing? {:margin-top 16}))
-          :uppercase? (get-in platform-specific [:discover :uppercase-subtitles?])
-          :font       :medium}
-    (label label-kw)]])
 
-(defview discover-popular [{:keys [contacts current-account]}]
+(defview top-status-for-popular-hashtag [{:keys [tag current-account]}]
+  (letsubs [discoveries [:get-popular-discoveries 1 [tag]]]
+    [view (merge styles/popular-list-container
+                 (get-in platform/platform-specific [:component-styles :discover :popular]))
+     [view styles/row
+      [view {}
+       [touchable-highlight {:on-press #(do (dispatch [:set :discover-search-tags [tag]])
+                                            (dispatch [:navigate-to :discover-search-results]))}
```

Comment:
@janherich might have some thoughts on this double dispatch

Reply:
@oskarth As do I, in fact. But I don't want that change to creep into this PR - it should be a separate refactor. That call is also an example of what is almost imperative programming. Since `navigate-to` doesn't accept and pass parameters, we first set a global mutable var and then call the view function that consumes it. Sure, we could do that in the same dispatch, but I can't stop thinking we can do better than that altogether.

Reply:
Ok fair, let's punt on it

Reply:
I agree with @goranjovic, what we do there is bad, but I let's refactor that action sequence into one nice single function with event hook in separate refactoring PR.
For now, please add a `TODO` comment there, so we won't forget about that :)

Reply:
> Since navigate-to doesn't accept and pass parameters
https://github.com/status-im/status-react/blob/d9db548e578575617a7d74fabf673456c2eefd15/src/status_im/ui/screens/navigation.cljs#L29
You can use `preload-data!` multimethod and pass data with `:navigate-to`. Probably `preload-data!` is not the best name though.